### PR TITLE
Cache the crates index and package cache for jobs that build rust

### PR DESCRIPTION
**Pull request by elliottt**
Rework the `install-rust` action to include caching of the crates index and cached downloaded crates.

### CODE REVIEW

**Review by jameysharp — APPROVED** [review]
I guess I wrote the action you're adding here, so it's not surprising that it looks reasonable to me. :laughing:

**Comment by alexcrichton** [comment]
Could this action be folded into the `install-rust` action, or invoked from that action, to avoid needing to specify both installation/caching?

**Comment by elliottt** [comment]
> Could this action be folded into the `install-rust` action, or invoked from that action, to avoid needing to specify both installation/caching?

It definitely could, though we'll need to translate the composite action to a javascript one.

**Comment by jameysharp** [comment]
Or rewrite `install-rust` as a composite action, which I claim is simpler than the JavaScript version: https://github.com/fastly/js-compute-runtime/blob/main/.github/actions/install-rust/action.yml

File: .github/actions/install-rust/action.yml [hunk 1]
```diff
@@ -1,12 +1,92 @@
...
+    required: false
+    default: '**/Cargo.lock'
 
 runs:
-  using: node16
-  main: 'main.js'
+  using: composite
+  steps:
+
+    # install rust
```
Comment (elliottt) [inline .github/actions/install-rust/action.yml:18]: What a helpful comment I added here.

**Review by alexcrichton — APPROVED** [review]

**Pull request merged**